### ISSUE TITLE
Web server doesn't support IPv6 when using the Arduino framework

### ISSUE THREAD

**Issue by dokutan** [esphome/issues#5095]
### The problem

The web server doesn't listen on IPv6 when using the arduino framework, esp-idf does work. I confirmed that the port is closed using nmap on the global and link local address.

### Which version of ESPHome has the issue?

2023.10.6

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

2023.10.1

### What platform are you using?

ESP32

### Board

esp32dev

### Component causing the issue

web_server

### Example YAML snippet

```yaml
esphome:
  name: test

esp32:
  board: esp32dev
  framework:
    type: arduino

network:
  enable_ipv6: true

web_server:
  port: 80
  local: true

sensor:
  - platform: uptime
    name: Uptime Sensor
```

**Comment by HeMan** [comment]
Could you test with the dev branch?

**Comment by HeMan** [comment]
Resolved in 2024.3.0
